File: static/views/order.js
let orderView = {
    renderOrder: function(data) {
        let content = document.querySelector(".booking-content");
        content.innerHTML = `
            <div class="booking-content__name">訂單編號 ${data.number}</div>
            <div class="booking-content__item">日期：<span>${data.trip.date}</span></div>
            <div class="booking-content__item">時間：<span>${data.trip.time}</span></div>
            <div class="booking-content__item">費用：<span>新台幣 ${data.price} 元</span></div>
            <div class="booking-content__item">地點：<span>${data.trip.attraction.name}</span></div>
            <div class="booking-content__item">付款狀態：<span>已完成付款</span></div>
        `;
    }
};

export default orderView;

File: views/order.py
from pydantic import BaseModel, EmailStr
from views.attraction import AttractionBase

class Contact(BaseModel):
    name: str
    email: EmailStr
    phone: str

class Trip(BaseModel):
    attraction: AttractionBase
    date: str
    time: str

class OrderBase(BaseModel):
    price: int
    trip: Trip
    contact: Contact

class Order(BaseModel):
    prime: str
    order: OrderBase

class PaymentBase(BaseModel):
    status: int
    message: str

class PaymentData(BaseModel):
    number: str
    payment: PaymentBase

class PaymentResult(BaseModel):
    data: PaymentData

class OrderPaidData(OrderBase):
    number: str
    status: int

class OrderPaid(BaseModel):
    data: OrderPaidData | None

class AttBase(BaseModel):
    id: int
    name : str
    address: str

class BookingBase(BaseModel):
    id: int
    attraction: AttBase
    date: str
    time: str

class OderAllBase(BaseModel):
    number: str
    price: int
    booking: BookingBase
    contact: Contact
    status: int
    created_time: str

class OderAll(BaseModel):
    data: list[OderAllBase] | None